lib/Conversion/ExportVerilog/LegalizeNames.cpp

```diff
@@ -199,7 +202,8 @@ static void legalizeModuleLocalNames(HWModuleOp module,
 /// Construct a GlobalNameResolver and do the initial scan to populate and
 /// unique the module/interfaces and port/parameter names.
 GlobalNameResolver::GlobalNameResolver(mlir::ModuleOp topLevel,
-                                       const LoweringOptions &options) {
+                                       const LoweringOptions &options)
+    : globalNameResolver(NameCollisionResolver(options)) {
```

Comment:
Nit, could this be:
```suggestion
    : globalNameResolver(options) {
```

Don't feel strongly, but also consider stashing the LoweringOptions reference as a member variable in this struct (`GlobalNameResolver`) instead of pulling it out of the `globalNameResolver` in the member functions.

Reply:
This would be better. 👍